website/docs/cloud-docs/integrations/kubernetes/index.mdx

```diff
@@ -1,68 +1,163 @@
+    version: 1.0.0
+  workspace:
+    name: workspace-name
+  variables:
+  - name: string_length
+  outputs:
+  - name: random_string
+```
 
-1. To create a Terraform workspace, you can create a separate Helm chart to deploy the custom resource or examine these [examples](https://github.com/hashicorp/terraform-helm/tree/master/example).
+The operator passes the `string_length` variable that exists in the workspace to the module and stores the `random_string` outputs as either a Kubernetes secret or a ConfigMap. If the output is marked as "sensitive", it is stored as a Kubernetes secret, otherwise it is stored as a ConfigMap. The variables need to be accessible within the workspace as a workspace variable, workspace variable set, or project variable set.
 
-### Upgrading
+Refer to the [module API reference](/terraform/cloud-docs/integrations/kubernetes/api-reference#module) for complete documentation of the `Module` specification.
 
-When a new version of the Terraform Cloud Operator for Kubernetes Helm Chart is available from the HashiCorp Helm repository, it can be upgraded with the following command:
+### Workspace
 
+Workspaces in Terraform Cloud are used to organize and manage Terraform configurations. The Terraform Cloud Operator allows you to create, configure, and manage workspaces directly from Kubernetes.
+
+The following example creates a new workspace named `us-west-development` configured to use Terraform version `1.6.2`. This workspace has two variables, `nodes` and `rds-secret`. The variable `rds-secret` is treated as sensitive, and the operator will read the value for the variable from a Kubernetes secret named `us-west-development-secrets`.
+
+```yaml
+---
+apiVersion: app.terraform.io/v1alpha2
+kind: Workspace
+metadata:
+  name: us-west-development
+spec:
+  organization: kubernetes-operator
+  token:
+    secretKeyRef:
+      name: tfc-operator
+      key: token
+  name: us-west-development
+  description: US West development workspace
+  terraformVersion: 1.6.2
+  applyMethod: auto
+  agentPool:
+    name: ap-us-west-development
+  terraformVariables:
+    - name: nodes
+      value: 2
+    - name: rds-secret
+      sensitive: true
+      valueFrom:
+        secretKeyRef:
+          name: us-west-development-secrets
+          key: rds-secret
+  runTasks:
+    - name: rt-us-west-development
+      stage: pre_plan
 ```
-helm upgrade --namespace ${RELEASE_NAMESPACE} ${RELEASE_NAME} hashicorp/terraform
-```
+
+In the above example, the `applyMethod` has the value of `auto`, so Terraform Cloud will automatically apply any changes to this workspace. The specification also configures the workspace to use the `ap-us-west-development` agent pool and run the `rt-us-west-development` run task at the `pre_plan` stage.
+
+The operator will store the value of the workspace outputs as Kubernetes secrets or ConfigMaps. If the outputs are marked as "sensitive", they are stored as Kubernetes secrets, otherwise they are stored as ConfigMaps.
```

Comment:
Since this is a pattern you've done twice, is this a good practice for folks to follow? Maybe we could explicitly state that? 

```suggestion
The operator stores the value of the workspace outputs as Kubernetes secrets or ConfigMaps. If the outputs are marked as "sensitive", they are stored as Kubernetes secrets, otherwise they are stored as ConfigMaps.
```